api/src/main/kotlin/edu/wgu/osmt/richskill/RichSkillController.kt

```diff
@@ -418,7 +388,7 @@ class RichSkillController @Autowired constructor(
         @PathVariable uuid: String
     ): HttpEntity<List<AuditLog>> {
         val pageable = OffsetPageable(0, Int.MAX_VALUE, AuditLogSortEnum.forValueOrDefault(AuditLogSortEnum.DateDesc.apiValue).sort)
-        val skill = richSkillRepository.findByUUID(uuid)
+        val skill = richSkillRepository.findByUUID(uuid) ?: throw ResponseStatusException(NOT_FOUND, "Skill with id $uuid not ready or not found")
```

Comment:
Ideally, I would push this kind of throw into the method signature for findByUUID (also for Collections above).